docs/concepts/about_biome.md

```diff
@@ -1,17 +1,415 @@
+available to an application.
+
+![alt_text](../images/biome_architecture_diagram.png "Biome architecture")
+
+### Local Storage for User, Key, and Credential Information
+
+Biome is local to a Splinter node &mdash; it does not share any user,
+credential, or key information with other Splinter nodes. Biome stores user,
+key, and credential information in a local PostgreSQL database.
+
+See [Biome database tables](#biome-database-tables) for the database structure
+and table names.
+
+### Key Management in Biome
+
+Biome's user key management provides the capability to store and retrieve
+public/private key pairs and associate them with a user. The Biome
+implementation is a general solution that reduces the overhead of writing a new
+key management system for any application that needs keys. It is designed to be
+flexible enough to support a multitude of use cases in an enterprise
+environment.
+
+The design of Biome includes the following assumptions:
+
+* The front-end application is responsible for encrypting the user's private key
+  (which is strongly recommended).
+
+* Biome does not enforce any correctness of the key pairs. The private key
+  should never be available to Biome in plain text, which makes it impossible to
+  check that a private key is actually associated with the public key.
+
+* Biome does not enforce any uniqueness of the key pairs. Though not ideal, this
+  design supports organizational structures that associate keys with specific
+  permissions rather than individual users.
+
+**Key management endpoints**
+
+The `splinterd` daemon exposes a set of endpoints that allow an application to
+add new key records, delete existing key records, fetch an existing key record,
+or update the display name of a key record. For more information, see [REST API
+Endpoints for Biome](#rest-api-endpoints-for-biome).
+
+**Key management library code**
+
+The CanopyJS library provides code so that Saplings (Splinter web apps) can use
+the key management functionality of Biome, as well as handlers to make key data
+available across Saplings running in Canopy. This code includes:
+
+* Function to generate a new keypair using the secp256k1 algorithm
+* Handlers to store an active key in Canopy state or fetch the currently active
+  key
+
+For example, the Profile Sapling uses Biome and the CanopyJS key management
+library to allow an app to take the following actions:
+
+*   Update user credentials (username or password)
+*   Create, delete, and view a given user's keys
+*   Select a key as the active key
+
+The Profile Sapling encrypts the private key before sending it to Biome. This
+Sapling is responsible for all direct communication with the Biome endpoints.
+
+### The libsplinter biome module
+
+For Rust developers, the `biome` module in the `libsplinter` library provides
+provides the following submodules, which operate independently:
+
+* [`credentials`](https://docs.rs/splinter/0.3.15/splinter/biome/credentials/index.html):
```

Comment:
The version in the URL is going to get outdated really quickly.

Reply:
Alas, yes.  I will research if it's possible to provide a "release-generic" link to a specific submodule's index.html.

Reply:
Changed to `/latest/` instead of `/0.3.15/`.  Note that these direct-to-content links are  used only for major items like modules and top-level submodules that are named in the docs.

Reply:
Changed to include `/latest` in the path instead of the version number.